retrofit/src/main/java/retrofit2/HttpServiceMethod.java

```diff
@@ -53,6 +53,12 @@ static <ResponseT, ReturnT> HttpServiceMethod<ResponseT, ReturnT> parseAnnotatio
         responseType = Utils.getParameterUpperBound(0, (ParameterizedType) responseType);
         continuationWantsResponse = true;
       } else {
+        if (getRawType(responseType) == Call.class) {
+          throw methodError(
+              method,
+              "Unable to create call adapter for suspend function with Call return type.");
```

Comment:
I would like this message to be a bit more instructive.

How about

> Suspend functions should not return Call, as they already execute asynchronously.
>
> Change its return type to ${..}

Where the expression is the parameterized type of Call.

Reply:
Make sense!